src/WorldState/WorldStateClient.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+using WorldState.Data.Models;
+
+namespace WorldState
+{
+    public class WorldStateClient : IDisposable
+    {
+        public Platform Platform { get; }
+
+        internal HttpClient HttpClient { get; }
+
+        private JsonSerializer serializer;
+        private UriBuilder     builder;
+        private string         platformAsString;
+
+        public WorldStateClient(Platform platform)
+        {
+            HttpClient = new HttpClient();
+            Platform   = platform;
+
+            builder          = new UriBuilder(WorldStateEndpoints.Base);
+            platformAsString = PlatformToString(platform);
+            serializer       = JsonSerializer.CreateDefault();
+        }
+
+        public WorldStateClient(Platform platform, Uri baseUrl)
+        {
+            if (baseUrl == null) throw new ArgumentNullException(nameof(baseUrl));
+
+            HttpClient = new HttpClient();
+            Platform   = platform;
+
+            builder          = new UriBuilder(baseUrl);
+            platformAsString = PlatformToString(platform);
+            serializer       = JsonSerializer.CreateDefault();
+        }
+
+        public WorldStateClient(Platform platform, Uri baseUrl, HttpMessageHandler messageHandler)
+        {
+            if (baseUrl == null) throw new ArgumentNullException(nameof(baseUrl));
+
+            HttpClient = new HttpClient(messageHandler);
+            Platform   = platform;
+
+            builder          = new UriBuilder(baseUrl);
+            platformAsString = PlatformToString(platform);
+            serializer       = JsonSerializer.CreateDefault();
+        }
+
+        #region IDisposable
+
+        protected virtual void Dispose(bool isDisposing)
```

Comment:
These are just auto my auto code-snippet for `IDisposable`. I think it was taken from a discussion on `IDisposable` best practice long ago and I've been using them since. Can you explain or just point me with a link why an explicit dtor is required?

Personally I don't like `sealed` anything. I'd like OSS projects to keep things open unless there's a security/vuneralbility concern. I guess it's up for discussion.

Reply:
I just looked it up and apparently the finalizer isn't needed. I apologize. However, understanding the pattern is still a good idea. [Microsoft has a good article about it.](https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose)

The benefit to keeping the class `sealed` is basically just simplicity I guess. If you're extracting an interface from this class, then anybody can write their own implementations of `IWorldStateClient` and even just pass a `WorldStateClient` to their implementation to use the methods defined here. It's up to you, I don't think it would make much of a difference.